packages/clay-date-picker/src/DayNumber.tsx

```diff
@@ -7,43 +7,84 @@ import {Keys} from '@clayui/shared';
 import classnames from 'classnames';
 import React from 'react';
 
-import {IDay, formatDate, setDate} from './Helpers';
+import {IDay, setDate} from './Helpers';
+import {TDaysSelected} from './types';
 
 interface IProps {
 	day: IDay;
-	daySelected: Date;
+	daysSelected: TDaysSelected;
 	disabled?: boolean;
-	onClick: (date: Date) => void;
+	range?: boolean;
+	onClick: (date: Date, range?: boolean) => void;
+}
+
+interface IInterval {
+	start: Date;
+	end: Date;
+}
+
+function isWithinInterval(dirtyDate: Date, dirtyInterval: IInterval) {
+	const interval = dirtyInterval || {};
+	const time = dirtyDate.getTime();
+	const startTime = interval.start.getTime();
+	const endTime = interval.end.getTime();
+
+	if (!(startTime <= endTime)) {
+		return false;
+	}
+
+	return time >= startTime && time <= endTime;
 }
 
 const ClayDatePickerDayNumber: React.FunctionComponent<IProps> = ({
 	day,
-	daySelected,
+	daysSelected,
 	disabled,
 	onClick,
+	range,
 }) => {
 	const {date, outside} = day;
+	const [startDate, endDate] = daysSelected;
+
+	const isSelectedToDate = date.toDateString() === endDate.toDateString();
+	const isSelectedFromDate = date.toDateString() === startDate.toDateString();
 
 	const classNames = classnames(
 		'date-picker-date date-picker-calendar-item',
 		{
-			active: date.toDateString() === daySelected.toDateString(),
+			active: isSelectedFromDate || (range && isSelectedToDate),
```

Comment:
Yes, it can be just `isSelectedFromDate ` :)
<img width="346" alt="Screen Shot 2021-04-13 at 17 36 02" src="https://user-images.githubusercontent.com/7663513/114617631-c3d10580-9c7e-11eb-9b1d-1e818d48ced6.png">

Another improvement is from: `hasStartDateSelected && !(hasEndDateSelected && hasStartDateSelected),` we can just `a && !hasEndDateSelected,` :)
<img width="341" alt="Screen Shot 2021-04-13 at 17 39 47" src="https://user-images.githubusercontent.com/7663513/114618022-40fc7a80-9c7f-11eb-96f0-447437f93374.png">

Reply:
I'm not good at boole's algebra for now but I use: https://www.dcode.fr/boolean-expressions-calculator for simplifying :)

Reply:
Result:

<img width="1539" alt="Screen Shot 2021-04-13 at 17 43 04" src="https://user-images.githubusercontent.com/7663513/114618429-b6684b00-9c7f-11eb-9571-785ae4d42f20.png">

Reply:
I was not referring to whether the conditional was right or not but I had the impression that `active` was only to be added when it is not a range but we need it regardless. Glad you managed to improve.

@pat270 I probably got it wrong here when you were saying it was different but I had a feeling that the `c-selected` would do the same as `active`, sorry for the confusion!! now it makes more sense 😅